pkg/reconciler/v1alpha1/autoscaling/kpa_scaler_test.go

```diff
@@ -50,14 +50,33 @@ func TestKPAScaler(t *testing.T) {
 		wantState     v1alpha1.RevisionServingStateType
 		wantReplicas  int
 		wantScaling   bool
+		kpaMutation   func(*kpa.PodAutoscaler)
 	}{{
-		label:         "scales to zero",
-		startState:    v1alpha1.RevisionServingStateActive,
+		label:         "waits to scale to zero",
+		startState:    v1alpha1.RevisionServingStateReserve,
 		startReplicas: 1,
 		scaleTo:       0,
 		wantState:     v1alpha1.RevisionServingStateReserve,
 		wantReplicas:  1,
 		wantScaling:   false,
+		kpaMutation: func(kpa *kpa.PodAutoscaler) {
+			kpa.Status.MarkInactive("foo", "bar")
+		},
+	}, {
+		label:         "scale to zero after grace period",
+		startState:    v1alpha1.RevisionServingStateReserve,
+		startReplicas: 1,
+		scaleTo:       0,
+		wantState:     v1alpha1.RevisionServingStateReserve,
+		wantReplicas:  0,
+		wantScaling:   true,
+		kpaMutation: func(k *kpa.PodAutoscaler) {
+			k.Status.Conditions = []kpa.PodAutoscalerCondition{{
+				Type:   "Active",
+				Status: "False",
+				// No LTT == a long long time ago
```

Comment:
Could you move this comment above line 74 to say 

```
// Not using MarkActive to leave LastTransitionTime untouched --> will be a long long time ago
```

Reply:
I added a test that checks the `gracePeriod` and just before.